versioned/transfer/README.md

```diff
@@ -0,0 +1,97 @@
+# Nessie export/import
+
+Functionality to export a Nessie repository and later import into another, empty Nessie repository.
+
+## Building blocks
+
+* Export functionality, based
+  on [AbstractNessieExporter](./src/main/java/org/projectnessie/versioned/transfer/AbstractNessieExporter.java)
+  to dump commits, named references, heads+fork points.
+* Import functionality, based
+  on [AbstractNessieImporter](./src/main/java/org/projectnessie/versioned/transfer/AbstractNessieImporter.java)
+  to load the exported data.
+* Commit log optimization to
+  * populate the list of parent-commits in all commits, according to the target Nessie repository's
+    configuration
+  * populate the key-lists in the commits, according to the target Nessie repository's configuration
+
+## Code examples
+
+```java
+class CodeExamples {
+  void exportExample() {
+    Path exportZipFile;
+
+    DatabaseAdapter databaseAdapter;
+
+    ZipArchiveExporter.builder()
+      .outputFile(exportZipFile)
+      .databaseAdapter(databaseAdapter)
+      .build()
+      .exportNessieRepository();
+  }
+
+  void importExample() {
+    Path exportZipFile;
+
+    DatabaseAdapter databaseAdapter;
+
+    ImportResult importResult =
+      ZipArchiveImporter.builder()
+        .sourceZipFile(exportZipFile)
+        .databaseAdapter(importDatabaseAdapter)
+        .build()
+        .importNessieRepository();
+
+    CommitLogOptimization.builder()
+      .headsAndForks(importResult.headsAndForkPoints())
+      .databaseAdapter(databaseAdapter)
+      .build()
+      .optimize();
+  }
+}
+```
+
+`ZipArchiveImporter` can be replaced with `FileImporter`.
+
+## Export contents
+
+Each export contains this information:
+
+* All commits (no specific order)
+* All named references including their heads
+* Heads + fork-points (used to feed commit-log optimization ran after a repository import)
+* Summary and inventory
+
+## `Content`, `CommitMeta`, global state, et al
+
+A Nessie export contains all `Content` information without any database internal information. This
+means that there is no information contained whether the source repository stored `Content` using
+e.g. global state. All `Content`s and `CommitMeta` are exported in their public JSON representation.
```

Comment:
Aren't we exporting this in protobuf serialization (byte array)? I didn't get the context for "JSON representation"

Reply:
They are exported as JSON in protobuf.